### PR TITLE
Feature/testing

### PR DESCRIPTION
**Pull request by isaka-lumato**
## UNIT TESTiNG

- created tests for all the possible behaviours
- followed good practices for testing

### CODE REVIEW

File: spec/student_spec.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  context 'When providing student information' do
+    student = Student.new(age: 20, name: 'Taro')
+
+    it 'Creates a new student with an age of 20 and a name of Taro' do
+      expect(student).to be_an_instance_of Student
+    end
+
+    it 'the student can have permission to use services' do
+      expect(student.can_use_services?).to eq true
+    end
```
Comment (mikethreels) [inline spec/student_spec.rb]: - [x] try to be more thorough when testing, for example in this case test positive and negative outcomes as well.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Good job on writing the tests they look good and are all passing 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/student_spec.rb (hunk 1 above)
Comment (Tchilo) [reply spec/student_spec.rb]: thanks will surely do that.

**Review by petudeveloper — APPROVED** [review]
Hi @Tchilo @isaka-lumato @yahyaelganyni1 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

bravo

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
